src/thread_pool.rs

```diff
@@ -430,16 +424,18 @@ impl ThreadPool {
         LOCK_LATCH.with(|latch| {
             let mut result = None;
             let job = StackJob::new(|| {
-                // SAFETY: Since this is within a job, and jobs only execute on
-                // worker threads, this must be non-null.
-                let worker_thread = unsafe { &*WorkerThread::current() };
-
-                // Run the user-provided function and write the output directly
-                // to the result.
-                result = Some(f(worker_thread, true));
-
-                // SAFETY: This latch is static, so the pointer is always valid.
-                unsafe { Latch::set(latch) };
+                WorkerThread::with(|worker_thread| {
+                    // Since this is within a job, and jobs only execute on
+                    // worker threads, this must be non-null.
+                    let worker_thread = worker_thread.unwrap();
```

Comment:
I appreciate that you kept the safety comments as non-panic comments on the `unwraps`.